src/main/java/Bro.java

```diff
@@ -0,0 +1,182 @@
+import bro.*;
+
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class Bro {
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+    private Parser parser;
+
+    /**
+     * Initializes the application components and starts the main event loop to handle user input.
+     * It sets up the user interface, parser, task list, and storage, then continuously listens for user commands
+     * until the user decides to exit by typing "bye".
+     *
+     * The method handles various commands:
+     * - "list" to display all tasks.
+     * - "mark <task number>" to mark a task as completed.
+     * - "unmark <task number>" to mark a task as not completed.
+     * - "delete <task number>" to delete a task.
+     * - "find <word> to find any tasks that contain that particular keyword.
+     * - "todo <task description>" to add a new to-do task.
+     * - "deadline <task description> /by <date time>" to add a task with a deadline.
+     * - "event <task description> /from <start date time> /to <end date time>" to add an event.
+     *
+     * The state of tasks is saved to a file after every change.
+     * Handles exceptions for file loading errors, command format errors, and task index out-of-bounds.
+     */
```

Comment:
I like how this description provides a clear and comprehensive overview of the application's functionality.